value/listreflect.go

```diff
@@ -40,3 +40,33 @@ func (r listReflect) Unstructured() interface{} {
 	}
 	return result
 }
+
+func (r listReflect) Range() ListRange {
+	return &listReflectRange{r.Value, newTempValuePooler(), -1, r.Value.Len()}
+}
+
+type listReflectRange struct {
+	val    reflect.Value
+	pooler *tempValuePooler
+	i      int
+	length int
```

Comment:
Very important optimization: you can make this a few bytes smaller by removing `length` since it can be computed from value?

Reply:
Done. I checked and computing length appears to be a cheap O(1) for all paths, so this should be fine.